webboot/webboot_test.go

```diff
@@ -0,0 +1,136 @@
+	expectedName string
+}
+
+func TestParseArg(t *testing.T) {
+	tests := []test{
+		{name: "Debian", linkOrName: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedLink: "https://cdimage.debian.org/debian-cd/current/amd64/iso-cd/debian-10.0.0-amd64-netinst.iso", expectedName: "debian-10.0.0-amd64-netinst.iso"},
+		{name: "TinyCore", linkOrName: "tinycore", expectedLink: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "tinycore"},
+		{name: "Ubuntu", linkOrName: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedLink: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu-18.04.2-desktop-amd64.iso"},
+	}
+
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			link, filename, err := parseArg(v.linkOrName)
+			if err != nil {
+				t.Fatal(err)
+			}
+			if strings.Compare(link, v.expectedLink) != 0 {
+				t.Errorf("Looking up URL for %v: got %v, want %v", v.linkOrName, link, v.expectedLink)
+			}
+			if strings.Compare(filename, v.expectedName) != 0 {
+				t.Errorf("Looking up the file name for %v: got %v, want %v", v.linkOrName, filename, v.expectedName)
+			}
+		})
+	}
+}
+
+func TestName(t *testing.T) {
+	tests := []test{
+		{name: "TinyCore", linkOrName: "http://tinycorelinux.net/10.x/x86_64/release/CorePure64-10.1.iso", expectedName: "CorePure64-10.1.iso"},
+		{name: "ArchLinux", linkOrName: "http://mirrors.edge.kernel.org/archlinux/iso/2019.08.01/archlinux-2019.08.01-x86_64.iso", expectedName: "archlinux-2019.08.01-x86_64.iso"},
+		{name: "Ubuntu", linkOrName: "http://releases.ubuntu.com/18.04.2/ubuntu-18.04.2-desktop-amd64.iso", expectedName: "ubuntu-18.04.2-desktop-amd64.iso"},
+		{name: "HtmlFile", linkOrName: "http://releases.ubuntu.com", expectedName: "index.html"},
+	}
+	for _, v := range tests {
+		t.Run(v.name, func(t *testing.T) {
+			filename, err := name(v.linkOrName)
+			if err != nil {
+				t.Fatalf("Looking up the file name for %v: got %v, want nil", v.linkOrName, err)
+			}
+			if strings.Compare(filename, v.expectedName) != 0 {
+				t.Errorf("Looking for name to generate for %v: got %v, want %v", v.linkOrName, filename, v.expectedName)
+			}
+		})
+	}
+
+}
+
+func TestWrite(t *testing.T) {
+	content := []byte("temporary file's content")
+	tmpDir, err := ioutil.TempDir("", "tmpDir")
+	if err != nil {
+		t.Fatalf("Failed to create a temporary directory")
+	}
+	tmpfile := filepath.Join(tmpDir, "tmpfile")
+	if err := ioutil.WriteFile(tmpfile, content, 0600); err != nil {
+		t.Fatal(err)
+	}
+
+	file, err := os.Open(tmpfile)
+	if err != nil {
+		t.Fatalf("Failed to open %v: got %v, want nil", tmpfile, err)
+	}
+
+	testfn := filepath.Join(tmpDir, "testfile")
+	err = write(file, testfn)
+
+	read, err := ioutil.ReadFile(testfn)
+	if !bytes.Equal(content, read) {
+		t.Fatalf("Failed to write to %v  properly: got %v, want %v", testfn, string(read), string(content))
```

Comment:
2 things: 
people will prefer that you not have 2 spaces before properly (I have gotten dinged on this)
and you don't need the "  properly" anyway. This is fine:
t.Fatalf("Writing %v: got %v, want %v", testfn, string(read), string(content))

Reply:
Sounds good, Ron! I got rid of the 'properly' and the 2 spaces, but I wanted to keep the error message t.Fatalf("Failed to write %v: got %v. want %v", testfn, string(read), string(content)). I like the style preference of keeping "failed to ...." :) However, I got rid of the 2 spaces and properly